### PR TITLE
Use AASM::NO_VALUE as `to_state` argument default value

### PR DESCRIPTION
**Pull request by teohm**
Based on discussion in https://github.com/aasm/aasm/issues/449#issuecomment-318022061

This PR attempts to fix the root cause of a few issues (#449, #416, #441) with this similar error:

```
Wrong number of arguments (given 1, expected 0)
```

This error happens because AASM incorrectly calls guard method with
one extra argument in certain scenarios. For example:

```ruby
def can_do_something?  # guard method with no argument
  ...
end

# But AASM incorrectly triggers `can_do_something?(nil)`
```

This happens because in `AASM::Core::Event#_fire(obj, options={}, to_state=nil, *args)`,
the `to_state` argument default nil value is causing ambiguity, i.e. there is no way for 
AASM to tell whether a caller provides a) one argument with nil value, or b) no argument.

This PR removes this ambiguity by using a symbol as `to_state` default value.

### CODE REVIEW

File: spec/unit/complex_multiple_example_spec.rb [hunk 1]
```diff
@@ -39,18 +39,18 @@
   it 'should be able to wait into waiting if polite' do
     auth.left_suspend!
     expect(auth.may_left_wait?(:waiting, :please)).to be true
-    auth.left_wait!(nil, :please)
```
Comment (teohm) [inline spec/unit/complex_multiple_example_spec.rb]: @anilmaurya I noticed `nil` is explicitly included in argument when triggering event in the specs. Is this a common in actual usage? This is different from the usage examples written in https://github.com/aasm/aasm/wiki/State-transitions-with-guard-clauses-that-require-arguments

Comment (anilmaurya) [inline spec/unit/complex_multiple_example_spec.rb]: @teohm Including `nil ` in argument list seems weird to me.

**Comment by anilmaurya** [comment]
Thank you @teohm for contributing , I appreciate your help 🎖

**Comment by teohm** [comment]
@anilmaurya By any chance to release this fix some time this month August? 🙏

**Comment by anilmaurya** [comment]
Hi @teohm ,

I am planning to release new version with this fix soon.

**Comment by teohm** [comment]
That's great! thanks @anilmaurya !! 🎉

**Comment by anilmaurya** [comment]
Hi @teohm , just release `4.12.2` with this fix.

**Comment by teohm** [comment]
Thanks @anilmaurya !!

**Pull request merged**